packages/repository/test/acceptance/repository.acceptance.ts

```diff
@@ -71,6 +71,36 @@ describe('Repository in Thinking in LoopBack', () => {
     expect(stored).to.containDeep({extra: 'additional data'});
   });
 
+  it('enables strict delete by default', async () => {
+    const p = await repo.create({slug: 'pencil'});
+    p.name = 'Red Pencil';
+    await repo.save(p);
+    await expect(repo.deleteById(10000)).to.be.rejectedWith(
+      /No instance with id/,
+    );
+  });
+
+  it('disables strict delete via configuration', async () => {
+    @model({settings: {strictDelete: false}})
+    class Pencil extends Entity {
+      @property({id: true})
+      id: number;
+      @property({type: 'string'})
+      name: string;
+    }
+
+    const pencilRepo = new DefaultCrudRepository<
+      Pencil,
+      typeof Pencil.prototype.id
+    >(Pencil, new DataSource({connector: 'memory'}));
+
+    await pencilRepo.create({
+      name: 'Green Pencil',
+    } as AnyObject);
```

Comment:
Why is it necessary to explicitly cast the value to `AnyObject`? In the test above, you called `repo.create({slug: 'pencil'})` with no type casts.